common/persistence/sql/workflow_state_non_maps_test.go

```diff
@@ -0,0 +1,180 @@
+    workflowID := "test"
+    runID := serialization.MustParseUUID("bf2360c6-a2c7-48a2-bd95-fe6254bca628")
+    testCases := []struct{
+        name string
+        signalRequestedIDs []string
+        deleteSignalRequestIDs []string
+        mockSetup func(*sqlplugin.MockTx)
+        wantErr bool
+    } {
+        {
+            name: "Success case",
+            signalRequestedIDs: []string{"s1", "s2"},
+            deleteSignalRequestIDs: []string{"s3", "s4"},
+            mockSetup: func(mockTx *sqlplugin.MockTx) {
+                mockTx.EXPECT().InsertIntoSignalsRequestedSets(gomock.Any(), []sqlplugin.SignalsRequestedSetsRow{
+                    {
+                        ShardID: int64(shardID),
+                        DomainID: domainID,
+                        WorkflowID: workflowID,
+                        RunID: runID,
+                        SignalID: "s1",
+                    },
+                    {
+                        ShardID: int64(shardID),
+                        DomainID: domainID,
+                        WorkflowID: workflowID,
+                        RunID: runID,
+                        SignalID: "s2",
+                    },
+                }).Return(nil, nil)
+                mockTx.EXPECT().DeleteFromSignalsRequestedSets(gomock.Any(), &sqlplugin.SignalsRequestedSetsFilter{
+                    ShardID: int64(shardID),
+                    DomainID: domainID,
+                    WorkflowID: workflowID,
+                    RunID: runID,
+                    SignalIDs: []string{"s3", "s4"},
+                }).Return(nil, nil)
+            },
+            wantErr: false,
+        },
+        {
+            name: "Error case - failed to insert",
+            signalRequestedIDs: []string{"s1", "s2"},
+            deleteSignalRequestIDs: []string{"s3", "s4"},
+            mockSetup: func(mockTx *sqlplugin.MockTx) {
+                err := errors.New("some error")
+                mockTx.EXPECT().InsertIntoSignalsRequestedSets(gomock.Any(), gomock.Any()).Return(nil, err)
+                mockTx.EXPECT().IsNotFoundError(err).Return(true)
+            },
+            wantErr: true,
+        },
+        {
+            name: "Error case - failed to delete",
+            signalRequestedIDs: []string{"s1", "s2"},
+            deleteSignalRequestIDs: []string{"s3", "s4"},
+            mockSetup: func(mockTx *sqlplugin.MockTx) {
+                err := errors.New("some error")
+                mockTx.EXPECT().InsertIntoSignalsRequestedSets(gomock.Any(), gomock.Any()).Return(nil, nil)
+                mockTx.EXPECT().DeleteFromSignalsRequestedSets(gomock.Any(), gomock.Any()).Return(nil, err)
+                mockTx.EXPECT().IsNotFoundError(err).Return(true)
+            },
+            wantErr: true,
+        },
+    }
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
```

Comment:
Finish is no longer required. Controller will be cleaned up with t.Cleanup.
See https://pkg.go.dev/github.com/golang/mock/gomock#Controller.Finish